src/System.Threading.Channels/tests/Stress.cs

```diff
@@ -0,0 +1,212 @@
+            foreach (bool allowSynchronousContinuations in new [] {false, true})
+            {
+                Func<ChannelOptions, Channel<int>> bounded = o => Channel.CreateBounded<int>((BoundedChannelOptions)o);
+                yield return new object[] { bounded, new BoundedChannelOptions(capacity)
+                                                        {
+                                                            SingleReader = singleReader,
+                                                            SingleWriter = singleWriter,
+                                                            AllowSynchronousContinuations = allowSynchronousContinuations,
+                                                            FullMode = bco
+                                                            }, readDelegate, writeDelegate
+                };
+            }
+        }
+
+        private static async Task<bool> ReadSynchronous(ChannelReader<int> reader)
+        {
+            while (!reader.TryRead(out int value))
+            {
+                if (!await reader.WaitToReadAsync())
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        private static async Task<bool> ReadASynchronous(ChannelReader<int> reader)
+        {
+            if (await reader.WaitToReadAsync())
+            {
+                await reader.ReadAsync();
+                return true;
+            }
+
+            return false;
+        }
+
+        private static async Task<bool> ReadSyncAndASync(ChannelReader<int> reader)
+        {
+            if (!reader.TryRead(out int value))
+            {
+                if (await reader.WaitToReadAsync())
+                {
+                    await reader.ReadAsync();
+                    return true;
+                }
+                return false;
+            }
+
+            return true;
+        }
+
+        private static async Task WriteSynchronous(ChannelWriter<int> writer, int value)
+        {
+            while (!writer.TryWrite(value))
+            {
+                if (!await writer.WaitToWriteAsync())
+                {
+                    break;
+                }
+            }
+        }
+
+        private static async Task WriteASynchronous(ChannelWriter<int> writer, int value)
+        {
+            if (await writer.WaitToWriteAsync())
+            {
+                await writer.WriteAsync(value);
+            }
+        }
+
+        private static async Task WriteSyncAndASync(ChannelWriter<int> writer, int value)
+        {
+            if (!writer.TryWrite(value))
+            {
+                if (await writer.WaitToWriteAsync())
+                {
+                    await writer.WriteAsync(value);
+                }
+            }
+        }
+
+        const int MaxNumberToWriteToChannel = 400_000;
+        private static readonly int MaxTaskCounts = Math.Max(2, Environment.ProcessorCount);
+
```

Comment:
Now that you've added the TestEnvironment thing, let's use it here.  Thanks for doing that.